### PR TITLE
Ingestor wallet.coinbase

### PR DESCRIPTION
**Pull request by pbkompasz**
### Functionality Supported
* Ingesting from URL: Yes
* Ingesting from Contract address: Yes
* Supported Networks: Base

### Before you submit
- [X] Ensure your generated MintTemplate works 😄
- [X] Ensure that your code is restricted to a single folder in `src/ingestors`
- [X] Ensure that all required assets are included (e.g. ABIs)
- [X] Ensure ABIs are trimmed to include only methods (1) used in the ingestor or (2) required to mint
- [X] Ensure that all exported methods are prefixed with the name of your ingestor e.g. `myMintingPlatformGetContractDetails`
- [X] Ensure that a test exists for generating a MintTemplate that will always succeed
- [X] Ensure that your code accesses **no external resources** except those passed in the `resources` object

### CODE REVIEW

File: src/ingestors/coinbase-wallet/index.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    return true;
+  }
+
+  async createMintForContract(
+    resources: MintIngestorResources,
+    contractOptions: MintContractOptions,
+  ): Promise<MintTemplate> {
+    const mintBuilder = new MintTemplateBuilder()
+      .setMintInstructionType(MintInstructionType.EVM_MINT)
+      .setPartnerName('Highlight');
```
Comment (agans) [inline src/ingestors/coinbase-wallet/index.ts:38]: hi! Looks like there are a couple highlight references in here

**Comment by pbkompasz** [comment]
@agans I removed the Highlight references and I also had to rewrite to fetch the collection metadata from the blockchain.

**Comment by pbkompasz** [comment]
@chrismaddern This is also ready for review.

**Comment by chrismaddern** [comment]
Will be reviewing this shortly (probably tomorrow vs. today)

Apologies for the delay. The typical review is not quick and most have led to rejections / changes so most are reviewed multiple times. Having to spread them out!! 🙏 🙏 🙏 🙏

**Comment by chrismaddern** [comment]
Hi @pbkompasz I pulled these to run tests locally before finalizing for merge.

I'm seeing one failure:
CleanShot 2024-08-26 at 16 42 15@2x

Could you take a look and get it passing?

It may be related to the fact that your `basicIngestorTests` aren't providing a static blockhash to simulate at - here's an example, and looks like you'll need to pull a blockhash from a couple weeks ago when you made this for it to work reliably moving forward.

e.g.
https://github.com/floornfts/mobile-minting/blob/main/test/ingestors/rodeo.test.ts#L30

**Review by chrismaddern — CHANGES_REQUESTED** [review]
Dropped a comment with changes at the top. Excited to get it across the line

**Review by chrismaddern — APPROVED** [review]

**Comment by chrismaddern** [comment]
> There might be some bug in the basicIngestorTests, because the simulation did fail in dry run but passed in basicIngestorTests.

I think it's because basicIngestorTests is using the blockhash provided, but dry-run is using the latest block hash?

Tests pass for me. Doing final review for merge! 🙂

**Pull request merged**